test/development/app-dir/ssr-in-rsc/ssr-in-rsc.test.ts

```diff
@@ -0,0 +1,544 @@
+/* eslint-disable jest/no-standalone-expect -- rule does not understand testGateWebpack */
+import { nextTestSetup } from 'e2e-utils'
+import {
+  assertHasRedbox,
+  assertNoRedbox,
+  getRedboxDescription,
+  getRedboxSource,
+  retry,
+} from 'next-test-utils'
+
+describe('react-dom/server in React Server environment', () => {
+  const dependencies = (global as any).isNextDeploy
+    ? // `link` is incompatible with the npm version used when this test is deployed
+      {
+        'internal-pkg': 'file:./internal-pkg',
+      }
+    : {
+        'internal-pkg': 'link:./internal-pkg',
+      }
+  const { next, isTurbopack } = nextTestSetup({
+    files: __dirname,
+    dependencies,
+    skipStart: true,
+  })
+
+  let isStarted = false
+  beforeEach(async () => {
+    // FIXME: next-custom-transforms RSC errors are not cleared during dev server livetime so we have to restart
+    if (isStarted) {
+      await next.stop()
+    }
+    await next.start()
+    isStarted = true
+  })
+
+  it('explicit react-dom/server.browser usage in app code', async () => {
+    const browser = await next.browser(
+      '/exports/app-code/react-dom-server-browser-explicit'
+    )
+
+    await assertNoRedbox(browser)
+    if (isTurbopack) {
+      expect(await browser.elementByCss('main').text()).toMatchInlineSnapshot(`
+        "{
+          "default": [
+            "renderToReadableStream",
+            "renderToStaticMarkup",
+            "renderToString",
+            "version"
+          ],
+          "named": [
+            "default",
+            "renderToReadableStream",
+            "renderToStaticMarkup",
+            "renderToString",
+            "version"
+          ]
+        }"
+      `)
+    } else {
+      expect(await browser.elementByCss('main').text()).toMatchInlineSnapshot(`
+        "{
+          "default": [
+            "renderToNodeStream",
+            "renderToReadableStream",
+            "renderToStaticMarkup",
+            "renderToStaticNodeStream",
```

Comment:
This is wrong. We need to update the APIs exposed from our optimized Webpack entrypoint.